packages/sdk/src/oracle/common/services/statemachine.ts

```diff
@@ -0,0 +1,266 @@
+import { Context, Memory, ContextType } from "../types/statemachine";
+import { ContextManager, shouldStep } from "../services/statemachines/utils";
```

Comment:
this file was copied from previous state machine file and modified to "register" different handler types. functionailty remains the same